### ISSUE TITLE
PowerShell ExecutionPolicy  errors 

### ISSUE THREAD

**Issue by sg1970** [walter-1/TSS#4]
Hello, 

I see a lot of ExecutionPolicy  errors ( listed at the end of this post) when I run the TSSv2.ps1 script (remotely using a admin account)  as follows:

`powershell -noprofile -ExecutionPolicy Unrestricted .\TSSv2.ps1 -SDP Perf -LogFolderName C:\Users\USER1\Desktop\TSSv2_Folder -AcceptEula $true *> .\TSSv2.txt`

I would like your help in getting a clean execution of the TSSv2 script without the Execution policy errors which are listed all the way down in this post.

This is the powershell log transcript as captured by the TSSv2.ps1 script and notice how the commandline as captured by TSSv2.ps1 is different from the actual command I typed(above). Its missing the  -ExecutionPolicy  param and therefore thats possibly whats causing the ExecutionPolicy errors IMHO ?  

Greatly appreciate your help on this. 

thanks!

********TSSv2 Transcript**************
Windows PowerShell transcript start
Start time: 20220525180641
Username: SVR1\USER1
RunAs User: SVR1\USER1
Machine: SVR1 (Microsoft Windows NT 10.0.14393.0)
Host Application: powershell -noprofile -ExecutionPolicy Unrestricted .\TSSv2.ps1 -SDP Perf -LogFolderName C:\Users\USER1\Desktop\TSSv2_Folder -AcceptEula $true *> .\TSSv2.txt

Process ID: 9032
PSVersion: 5.1.14393.5066
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.14393.5066
BuildVersion: 10.0.14393.5066
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
**********************
****20220525 18:06:41.991**** **Commandline**: .\TSSv2.ps1 -SDP Perf -LogFolderName C:\Users\USER1\Desktop\TSSv2_Folder -AcceptEula $true *> .\TSSv2.txt
20220525 18:06:42.023 ... running TSSv2 v2022.05.19.3 on OS: 10.0.14393.0 with PS version: 5.1
20220525 18:06:48.257 [psSDP] SDP(Perf) starting...
20220525 18:06:48.316 [psSDP] Running .\Get-psSDP.ps1 Perf -SkipEULA -skipQEdit -savePath C:\Users\USER1\Desktop\Test\TSSv2_SVR1_220525-180641_  -EvtDaysBack 30
**********************
Windows PowerShell transcript end
End time: 20220525180649
**********************

**---------**ERROR Message** ---------**
Get-BPAModel : There has been an error while trying to set the process wide execution policy. 
(Inner Exception: Windows PowerShell updated your execution policy successfully, but the setting 
is overridden by a policy defined at a more specific scope.  Due to the override, your shell will 
retain its current effective execution policy of Bypass. Type "Get-ExecutionPolicy -List" to view 
your execution policy settings. For more information please see "Get-Help Set-ExecutionPolicy".)
At line:1 char:1
+ Get-BPAModel
+ ~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-BpaModel], CommandLetException
    + FullyQualifiedErrorId : SetExecutionPolicyError,Microsoft.BestPractices.Cmdlets.GetModelCom 
   mand

**Comment by walter-1** [comment]
Hello,
sorry for late answer, I'm on vacation.
I cannot reproduce your errors. tested: `powershell -noprofile -ExecutionPolicy Unrestricted .\TSSv2.ps1 -SDP Perf -AcceptEula`

Maybe there is a Powershell execution policy mismatch on your Server 2016; did you verify with command "`Get-ExecutionPolicy -List`" ?

If the BPA error is the only error, you can append switch -`SkipSDPList skipBPA`
And please always use only latest TSS version from https://aka.ms/getTSS   (today's version is _2022.06.06.0_)

**Comment by sg1970** [comment]
What is BPA and is it ok to skip that functionality ?

But here is the output after skipping BPA its much cleaner but there is still an error ( bolded at the end)

I will make a new post to answer your question about Powershell execution policy mismatch on my server 

-----------

20220610 14:36:43.657 LogFolder is set to 
'C:\Users\User1\Desktop\MyApp\SVR1_20220610_1430\TSSv2\TSSv2_SVR1_220610-143643_'
20220610 14:36:43.829 ... running TSSv2 v2022.06.06.0 on OS: 6.3.9600.0 with PS version: 5.1
20220610 14:36:50.845 SkipOptionSwitchs = -skipBPA 
20220610 14:36:50.845 [psSDP] SDP(Perf) starting...
20220610 14:36:50.845 [psSDP] Running .\Get-psSDP.ps1 Perf -SkipEULA -skipQEdit -savePath 
C:\Users\User1\Desktop\MyApp\SVR1_20220610_1430\TSSv2\TSSv2_SVR1_220610-143643_ 
-skipBPA  -EvtDaysBack 30
EULA Accepted
14:36:52 starting psSDP v2022.05.31.0 Data Collection Report for Windows Technology: 'Perf'
** -runFull:  -SkipTS:  -SkipBPA: True  -skipNetView:  -skipSddcDiag:  -noNetAdapters:  -skipHVreplica:   -skipCsvSMB: 
 -skipHang:  -skipXray: 


14:36:52 _Copy utils...
14:37:03 _Copy utils... done
[2022-06-10T14:37:05] [Get-psSDP.ps1:2078]: [Starting Discovery]
[2022-06-10T14:37:05] [Get-psSDP.ps1:1832]: DiscoveryReport C:\Users\User1\Desktop\MyApp\SVR1
_20220610_1430\TSSv2\TSSv2_SVR1_220610-143643_\Results_Perf_220610-183651\SVR1_DiscoveryReport.xml 
does not exist. Creating a new report
[2022-06-10T14:37:05] [Get-psSDP.ps1:2084]: [Starting Discovery] End Discovery

... starting script: TS_Main_Perf.ps1 (... please be patient until all modules finished ...)
 --- 14:37:07 Perf - Collect Phase module: Common ---
   Write-DiagProgress: System Information: Running MSInfo32 utility to obtain system information (TXT and NFO)
   Write-DiagProgress: Process and Driver Information (pstat): Running pstat tool to obtain process information
   Write-DiagProgress: File Version Information (ChkSym): C:\Program Files\*.sys
   Write-DiagProgress: File Version Information (ChkSym): C:\Program Files (x86)\*.sys
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\drivers\*.*
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\*.DLL
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\SysWOW64\*.dll
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\*.EXE
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\SysWOW64\*.exe
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\*.SYS
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\SysWOW64\*.sys
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\Spool\*.*
   Write-DiagProgress: File Version Information (ChkSym): C:\Windows\System32\iscsi*.*
   Write-DiagProgress: File Version Information (ChkSym): Running Processes
   Write-DiagProgress: File Version Information (ChkSym): Running Drivers
   Write-DiagProgress: Autoruns: Obtaining currently configured auto-start applications and services
   Write-DiagProgress: Server Manager Information: Obtaining Server Manager logs and role information
   Write-DiagProgress: Schedule Tasks Information: Obtaining information about schedule tasks
   Write-DiagProgress: Event Logs: Exporting Event LogDays: 30
   Write-DiagProgress: Event Logs: Exporting Event Log: System
   Write-DiagProgress: Event Logs: Exporting Event Log: Application
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-ReFS/Operational
   Write-DiagProgress: Resultant Set of Policy (RSoP): Obtaining Resultant Set of Policy (RSoP) via gpresult.exe utility
   Write-DiagProgress: Collecting System Information: Collecting Data
   Write-DiagProgress: Collecting System Information: Formatting Data
   Write-DiagProgress: Information about driver verifier: Obtaining information about driver verifier (verifier.exe)
   Write-DiagProgress: Boot Information: Collecting Boot Information
   Write-DiagProgress: Information about devices and connections: Gathering devices and connections information (devcon)
   Write-DiagProgress: User Rights Diagnostic Tool: Running UserRights.exe to gather user rights assignments
   Write-DiagProgress: Event Logs: Exporting Event LogDays: 30
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-TaskScheduler/Operational
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-TaskScheduler/Debug
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-TaskScheduler/Diagnostic
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-TaskScheduler/Maintenance
   Write-DiagProgress: Whoami Diagnostic Tool: Running Whoami.exe to gather the user's SID, group memberships and privileges
   Write-DiagProgress: PoolMon information: Gathering PoolMon information (PoolMon.exe)
   Write-DiagProgress: Summary Report: Gathering Summary Report
   Write-DiagProgress: Registry Entries: Collecting KIR Known Issue Rollback registry entries.
   Write-DiagProgress: System and Software Registry Hives: Collecting Registry Hives.
 --- 14:39:59 Perf - Collect Phase module: HyperV  ---
 --- 14:40:00 Perf - Collect Phase module: DOM ---
   Write-DiagProgress: Kerberos tickets and TGT: Running KList utility to obtain Kerberos related information
 --- 14:40:00 Perf - Collect Phase module: Net ---
   Write-DiagProgress: SMB Server Component: Collects information about the SMB Server Component.
   Write-DiagProgress: SMB Server Component: net accounts
   Write-DiagProgress: SMB Server Component: net config server
   Write-DiagProgress: SMB Server Component: net session
   Write-DiagProgress: SMB Server Component: net files
   Write-DiagProgress: SMB Server Component: net share
   Write-DiagProgress: Event Logs: Exporting Event LogDays: 30
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-SMBServer/Connectivity
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-SMBServer/Operational
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-SMBServer/Security
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-SMBDirect/Operational
   Write-DiagProgress: TCPIP: Collects information about TCPIP.
   Write-DiagProgress: TCPIP: hostname 
   Write-DiagProgress: TCPIP: ipconfig /allcompartments /all
   Write-DiagProgress: TCPIP: route print 
   Write-DiagProgress: TCPIP: arp -a
   Write-DiagProgress: TCPIP: netstat -nato
   Write-DiagProgress: TCPIP: netstat -anob
   Write-DiagProgress: TCPIP: netstat -es
   Write-DiagProgress: TCPIP: netsh int tcp show global
   Write-DiagProgress: TCPIP: netsh int ipv4 show offload
   Write-DiagProgress: TCPIP: netstat -nato -p tcp
The WinNAT service is stopped. Please start the service to display configuration from active store.

The WinNAT service is stopped. Please start the service to display configuration from active store.

   Write-DiagProgress: TCPIP: netsh int tcp show global
   Write-DiagProgress: TCPIP: netsh int tcp show heuristics
   Write-DiagProgress: TCPIP: netsh int tcp show chimneyapplications
   Write-DiagProgress: TCPIP: netsh int tcp show chimneyports
   Write-DiagProgress: TCPIP: netsh int tcp show chimneystats
   Write-DiagProgress: TCPIP: netsh int tcp show netdmastats
   Write-DiagProgress: TCPIP: netsh int tcp show rscstats
   Write-DiagProgress: TCPIP: netsh int tcp show security
   Write-DiagProgress: TCPIP: netsh int tcp show supplemental
   Write-DiagProgress: TCPIP: netsh int tcp show supplementalports
   Write-DiagProgress: TCPIP: netsh int tcp show supplementalsubnets
   Write-DiagProgress: TCPIP: netsh int show int
   Write-DiagProgress: TCPIP: netsh int ipv4 show int
   Write-DiagProgress: TCPIP: netsh int ipv4 show addresses
   Write-DiagProgress: TCPIP: netsh int ipv4 show ipaddresses
   Write-DiagProgress: TCPIP: netsh int ipv4 show compartments
   Write-DiagProgress: TCPIP: netsh int ipv4 show dnsservers
   Write-DiagProgress: TCPIP: netsh int ipv4 show winsservers
   Write-DiagProgress: TCPIP: netsh int ipv4 show dynamicportrange tcp
   Write-DiagProgress: TCPIP: netsh int ipv4 show dynamicportrange udp
   Write-DiagProgress: TCPIP: netsh int ipv4 show global
   Write-DiagProgress: TCPIP: netsh int ipv4 show icmpstats
   Write-DiagProgress: TCPIP: netsh int ipv4 show ipstats
   Write-DiagProgress: TCPIP: netsh int ipv4 show joins
   Write-DiagProgress: TCPIP: netsh int ipv4 show offload
   Write-DiagProgress: TCPIP: netsh int ipv4 show route
   Write-DiagProgress: TCPIP: netsh int ipv4 show subint
   Write-DiagProgress: TCPIP: netsh int ipv4 show tcpconnections
   Write-DiagProgress: TCPIP: netsh int ipv4 show tcpstats
   Write-DiagProgress: TCPIP: netsh int ipv4 show udpconnections
   Write-DiagProgress: TCPIP: netsh int ipv4 show udpstats
   Write-DiagProgress: TCPIP: netsh int ipv4 show destinationcache
   Write-DiagProgress: TCPIP: netsh int ipv4 show ipnettomedia
   Write-DiagProgress: TCPIP: netsh int ipv4 show neighbors
   Write-DiagProgress: TCPIP: netsh int show int
   Write-DiagProgress: TCPIP: netsh int ipv6 show int
   Write-DiagProgress: TCPIP: netsh int ipv6 show addresses
   Write-DiagProgress: TCPIP: netsh int ipv6 show compartments
   Write-DiagProgress: TCPIP: netsh int ipv6 show destinationcache
   Write-DiagProgress: TCPIP: netsh int ipv6 show dnsservers
   Write-DiagProgress: TCPIP: netsh int ipv6 show dynamicportrange tcp
   Write-DiagProgress: TCPIP: netsh int ipv6 show dynamicportrange udp
   Write-DiagProgress: TCPIP: netsh int ipv6 show global
   Write-DiagProgress: TCPIP: netsh int ipv6 show ipstats
   Write-DiagProgress: TCPIP: netsh int ipv6 show joins
   Write-DiagProgress: TCPIP: netsh int ipv6 show neighbors
   Write-DiagProgress: TCPIP: netsh int ipv6 show offload
   Write-DiagProgress: TCPIP: netsh int ipv6 show potentialrouters
   Write-DiagProgress: TCPIP: netsh int ipv6 show prefixpolicies
   Write-DiagProgress: TCPIP: netsh int ipv6 show privacy
   Write-DiagProgress: TCPIP: netsh int ipv6 show route
   Write-DiagProgress: TCPIP: netsh int ipv6 show siteprefixes
   Write-DiagProgress: TCPIP: netsh int ipv6 show siteprefixes
   Write-DiagProgress: TCPIP: netsh int ipv6 show subint
   Write-DiagProgress: TCPIP: netsh int ipv6 show tcpstats
   Write-DiagProgress: TCPIP: netsh int ipv6 show teredo
   Write-DiagProgress: TCPIP: netsh int ipv6 show udpstats
   Write-DiagProgress: TCPIP: netsh int ipv6 show int
   Write-DiagProgress: TCPIP: netsh int 6to4 show int
   Write-DiagProgress: TCPIP: netsh int 6to4 show relay
   Write-DiagProgress: TCPIP: netsh int 6to4 show routing
   Write-DiagProgress: TCPIP: netsh int 6to4 show state
   Write-DiagProgress: TCPIP: netsh int httpstunnel show interfaces
   Write-DiagProgress: TCPIP: netsh int httpstunnel show statistics
   Write-DiagProgress: TCPIP: netsh int isatap show router 
   Write-DiagProgress: TCPIP: netsh int isatap show state 
   Write-DiagProgress: TCPIP: netsh int teredo show state 
   Write-DiagProgress: TCPIP: netsh int ipv6 show int level=verbose
   Write-DiagProgress: TCPIP: netsh netio show bindingfilters
   Write-DiagProgress: TCPIP: netsh int portproxy show all
   Write-DiagProgress: Event Logs: Exporting Event LogDays: 30
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-Iphlpsvc/Operational
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-WLAN-AutoConfig/Operational
 --- 14:40:58 Perf - Collect Phase module: Print ---
   Write-DiagProgress: Event Logs: Exporting Event LogDays: 30
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-PrintService/Admin
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-PrintService/Operational
   Write-DiagProgress: Event Logs: Exporting Event Log: Microsoft-Windows-PrintService/Debug
 --- 14:41:01 Perf - Collect Phase module: Storage ---
   Write-DiagProgress: Storage related event logs on System log: Exporting system log to obtain storage events
   Write-DiagProgress: Storage related event logs on System log: Searching for storage related event logs on system logs
 --- 14:41:03 Perf - Collect Phase module: WinRM ---
   Write-DiagProgress: File Version Information (ChkSym): File version information
 --- 14:41:04 Perf - Collect Phase module: Update ---
   Write-DiagProgress: Installed updates/hotfixes: Obtaining Update History
   Write-DiagProgress: Installed updates/hotfixes: Generating output files
   Write-DiagProgress: Installed updates/hotfixes: Generate Update Histories files and collect it
   Write-DiagProgress: Collecting Windows Update: Collecting \Windows\windowsupdate.log
 --- 14:41:11 Perf - Collect Phase module: Cluster ---
Machine is not a cluster
 --- 14:41:12 Perf - Collect Phase module: Performance Data ---
   Write-DiagProgress: System Performance Overview: Running Performance Monitor and collecting information for 1 min ...
   Write-DiagProgress: Network Adapters: Collects information about the Network Adapters
... running Get-NetAdapterStatistics, hint: use .\Get-psSDP Net -noNetAdapters, if stuck here
 --- 14:42:34 Perf - Collect Phase: Done ---
...Next step: Troubleshooting section, if it hangs, run script with parameter SkipTS
 --- 14:42:34 Perf - Diag Phase module: TS_common ---
   Write-DiagProgress: Memory Dump Related Issues: Collecting Memory Dump Related Information
   Write-DiagProgress: Checking information for Evaluation Media: 
   Write-DiagProgress: Global Flags and Application Verifier Checks: Obtaining Global Flags and Application Verifier settings
   Write-DiagProgress: Driver Verifier Check: Obtaining Verifier settings
   Write-DiagProgress: Processes Information: Obtaining basic performance information from running processes
   Write-DiagProgress: Checking Registry Size Limit: Checking if Registry Size Limit setting is present on the system
 --- 14:42:46 Perf - Diag Phase module: TS_Net ---
   Write-DiagProgress: TCP Port Usage: Obtaining TCP/IP port usage statistics
   Write-DiagProgress: Checking Open Files: Checking open files via server service
   Write-DiagProgress: Checking for KB978309: Checking for Tunnel.sys file
 --- 14:42:49 Perf - Diag Phase module: TS_HyperV ---
   Write-DiagProgress: Virtualization Information: Obtaining Information about Virtualization Environment
   Write-DiagProgress: Checking Event Logs: Checking event logs for known issues.
   Write-DiagProgress: Checking SCSI Subkeys for Hyper-V: Checking for Hyper-V SCSI Registry Subkeys
   Write-DiagProgress: Checking for KB2263829: Checking version of Hvax64.exe and other binaries
 --- 14:42:52 Perf - Diag Phase module: TS_Print ---
 --- 14:42:52 Perf - Diag Phase module: TS_RDP ---
   Write-DiagProgress: Checking the presence of Citrix AppSense: Checking if Citrix AppSense is installed on machine
   Write-DiagProgress: Checking for Inactive TS Ports: Checking for the presence of a large number of inactive Terminal Server ports.
 --- 14:42:52 Perf - Diag Phase module: TS_Storage ---
 --- 14:42:53 Perf - Diag Phase module: TS_Setup ---
This system is not up-to-date. Many known issues are resolved by applying latest cumulative update!
*** [WARNING] latest OS cumulative KB 5014011 is missing.
 Please update this machine with recommended Microsoft KB 5014011 and verify if your issue is resolved.
*** 14:43:50 DONE TS_AutoAddCommands_Perf.ps1 SkipTS:  - SkipBPA: True
xray by tdimli, v1.0.220527.0

Starting diagnostics, checking for known issues...
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.

17 diagnostic check(s) run (R:4 S:13 F:0)
0 issue(s) found
Diagnostics completed.

14:44:01 ... now zipping 'Perf' data folder C:\Users\User1\Desktop\MyApp\SVR1_20220610_1430\T
SSv2\TSSv2_SVR1_220610-143643_\Results_Perf_220610-183651
 [psSDP-Info] done. Resulting cabinet/zip: C:\Users\User1\Desktop\MyApp\SVR1_20220610_1430\TS
Sv2\TSSv2_SVR1_220610-143643_\tss_220610-184401_SVR1_psSDP_Perf.zip
**14:44:15 *** end of psSDP Report datacollection for Windows Technology: 'Perf'
20220610 14:44:16.019 [ProcessSDP(11368)] ERROR: An Exception happend in Get-psSDP.ps1
Command/Function: Set-ExecutionPolicy failed with 0x8013150a => Unknown error (0x8013150a)**
SecurityException: Security error.
ScriptStack:
at <ScriptBlock><Process>, C:\Users\User1\Desktop\MyApp\TSSv2\psSDP\Get-psSDP.ps1: line 955
at <ScriptBlock>, <No file>: line 1
at ProcessSDP, C:\Users\User1\Desktop\MyApp\TSSv2\TSSv2.ps1: line 11364
at <ScriptBlock>, C:\Users\User1\Desktop\MyApp\TSSv2\TSSv2.ps1: line 17695
at <ScriptBlock>, <No file>: line 1
20220610 14:44:16.034 [psSDP] end of SDP data collection

**Comment by walter-1** [comment]
BPA = Best Practice Analyzer
- the error at "psSDP\Get-psSDP.ps1: line 955" happens, because the script uses commands
`Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass -Force"` at beginning, and resets policy to original state at the end
`Set-ExecutionPolicy -Scope Process -ExecutionPolicy $StartExecPolicy -Force -ErrorAction SilentlyContinue`

**Comment by sg1970** [comment]
> 
> Maybe there is a Powershell execution policy mismatch on your Server 2016; did you verify with command "`Get-ExecutionPolicy -List`" ?


Here is the output of that command on my server:

```
PS C:\Windows\system32> Get-ExecutionPolicy -List

        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy          Bypass
   UserPolicy          Bypass
      Process       Undefined
  CurrentUser       Undefined
 LocalMachine    RemoteSigned



PS C:\Windows\system32> 
```

**Comment by sg1970** [comment]
New error that I noticed today while testing on a different server is included below. Any advice on how to avoid these errors will be greatly appreciated as these scripts are triggered from a automated process. Therefore must work for a variety of Servers

```

20220614 19:46:52.064 LogFolder is set to 'C:\Users\User1\Desktop\WTLESuppor
tScript\SVRTXXX12345_20220614_1940\TSSv2\TSSv2_SVRTXXX12345_220614-194652_'
20220614 19:46:52.252 ... running TSSv2 v2022.06.06.0 on OS: 10.0.14393.0 with PS version: 5.1
20220614 19:46:57.771 SkipOptionSwitchs = -skipBPA 
20220614 19:46:57.771 [psSDP] SDP(Perf) starting...
20220614 19:46:57.771 [psSDP] Running .\Get-psSDP.ps1 Perf -SkipEULA -skipQEdit -savePath C:\Users
\User1\Desktop\WTLESupportScript\SVRTXXX12345_20220614_1940\TSSv2\TSSv2_SVRT
000103AC_220614-194652_ -skipBPA  -EvtDaysBack 30
EULA Accepted
19:46:59 starting psSDP v2022.05.31.0 Data Collection Report for Windows Technology: 'Perf'
** -runFull:  -SkipTS:  -SkipBPA: True  -skipNetView:  -skipSddcDiag:  -noNetAdapters:  
-skipHVreplica:   -skipCsvSMB:  -skipHang:  -skipXray: 


19:46:59 _Copy utils...
C:\Users\User1\Desktop\WTLESupportScript\TSSv2\psSDP\Get-psSDP.ps1 : 
Get-psSDP.ps1 v2022.05.31.0 caught an exception on SVRTXXX12345:

Exception Type:    System.IO.DirectoryNotFoundException

Exception Message: Could not find a part of the path 'C:\Users\User1\Desktop
\WTLESupportScript\SVRTXXX12345_20220614_1940\TSSv2\TSSv2_SVRTXXX12345_220614-194652_\Results_Perf
_220614-234658\utils_Exchange_all_exchange_versions_using_powershell_should_not_be_called_while_lo
ading_powershell.ps1'.
At line:1 char:1
+ .\Get-psSDP.ps1 Perf -SkipEULA -skipQEdit -savePath C:\Users\WindowsH ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-psSDP.ps1
 
801
[ERROR] v2022.05.31.0 Please report above error message in Get-psSDP.ps1 line: 801 - Elevated: 
True

 [psSDP-Info] in case you see any red error messages regarding signing, [method#1] open an Admin 
Powershell CMD and run this command first:
	 Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass -Force
     [Alternate method#2]: To self sign .ps1 scripts: run in elevated CMD
         tss_PS1sign.cmd Get-psSDP
     [Alternate method#3a]: If scripts are blocked by MachinePolicy, run in elevated Powershell: 
	 Set-ItemProperty -Path HKLM:\Software\Policies\Microsoft\Windows\PowerShell -Name 
ExecutionPolicy -Value ByPass
	 Set-ItemProperty -Path HKLM:\Software\Policies\Microsoft\Windows\PowerShell -Name EnableScripts 
 -Value 1 -Type DWord
     [Alternate method#3b]: If scripts are blocked by UserPolicy, run in elevated Powershell: 
	 Set-ItemProperty -Path HKLM:\Software\Microsoft\PowerShell\1\ShellIds\Microsoft.PowerShell 
-Name ExecutionPolicy -Value ByPass
	 Set-ItemProperty -Path HKLM:\Software\Microsoft\PowerShell\1\ShellIds\Microsoft.PowerShell 
-Name EnableScripts  -Value 1 -Type DWord 

20220614 19:47:06.787 [ProcessSDP(11368)] ERROR: An Exception happend in Get-psSDP.ps1
Command/Function: Set-ExecutionPolicy failed with 0x8013150a => Unknown error (0x8013150a)
SecurityException: Security error.
ScriptStack:
at <ScriptBlock><Process>, 
C:\Users\User1\Desktop\WTLESupportScript\TSSv2\psSDP\Get-psSDP.ps1: line 955
at <ScriptBlock>, <No file>: line 1
at ProcessSDP, C:\Users\User1\Desktop\WTLESupportScript\TSSv2\TSSv2.ps1: 
line 11364
at <ScriptBlock>, C:\Users\User1\Desktop\WTLESupportScript\TSSv2\TSSv2.ps1: 
line 17695
at <ScriptBlock>, <No file>: line 1
20220614 19:47:06.787 [psSDP] end of SDP data collection
```

**Comment by walter-1** [comment]
-	I cannot reproduce the issue(s) reported above (as I have no Exchange server for testing)
-	For limited support, we would need the logfiles of all the different error situations (at least the three log files
-* __ErrorVariable.txt
*__Log-transcript.txt
*__Log-Warn-Err-Info.txt
- regarding latest error: Script should be started in a short folder like C:\TSSv2\, not using long paths (which might at the end exceed MAX_PATH limitation) I see: 'C:\Users\User1\Desktop\WTLESupportScript>
And when using switch **-FolderPath,** that path should be kept short as well, like C:\Data
Best practice (tested): extract TSS to C:\TSSv2, use similar short folder path when specifying switch **-FolderPath**

**Comment by walter-1** [comment]
**STATUS**: closing
further actions have been addressed in case 2206210060001260
